src/App.jsx

```diff
@@ -1,14 +1,50 @@
+// Wave 1: Import ChatEntry to show one message
+// import ChatEntry from './components/ChatEntry';
 import './App.css';
+import messages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
+const calculateTotalLikes = (chatData) => {
+  return chatData.reduce((total, message) => {
+    return total + (message.liked ? 1 : 0);
+  }, 0);
+};
 
 const App = () => {
+  // Wave 1: Use first message data to render one ChatEntry
+  // const chatMessages = messages[0];
+
+  const [chatData, setChatData] = useState(messages);
+  const totalLikes = calculateTotalLikes(chatData);
+
+  const toggleLike = (id) => {
+    setChatData((chatData) =>
+      chatData.map((entry) => {
+        if (entry.id === id) {
+          return { ...entry, liked: !entry.liked };
```

Comment:
We showed this approach in class, but technically, we're mixing a few responsibilities here. rather than this function needing to know how to change the liked status itself, we could move this update logic to a helper function. This would better mirror how we eventually update records when there's an API call involved.

In this project, our messages are very simple objects, but if we had more involved operations, it could be worthwhile to create an actual class with methods to work with them, or at least have a set of dedicated helper functions to centralize any such mutation logic.